comtypes/typelib_loader.py

```diff
@@ -0,0 +1,255 @@
+import sys
+from comtypes.client import GetModule
+import winreg
+
+# this makes it easy to load registered type libraries. It uses the .NET
+# names that are stored in the registry to function. Not all .NET names
+# can be used because there isn't typelib information available for all
+# of them.
+# Here are a few examples
+#
+# if you have an Nvidia video card
+# comtypes.typelib.DisplayServer.Config
+#
+# Windows Media Player
+# comtypes.typelib.MediaPlayer.MediaPlayer
+
+
+def _get_reg_value(path, key, wow6432=False):
+    d = _read_reg_values(path, wow6432)
+    if key in d:
+        return d[key]
+
+    return ''
+
+
+def _read_reg_keys(key, wow6432=False):
+    if isinstance(key, tuple):
+        root = key[0]
+        key = key[1]
+    else:
+        root = winreg.HKEY_CLASSES_ROOT
+
+    try:
+        if wow6432:
+            handle = winreg.OpenKey(
+                root,
+                key,
+                0,
+                winreg.KEY_READ | winreg.KEY_WOW64_32KEY
+            )
+        else:
+            handle = winreg.OpenKeyEx(root, key)
+    except winreg.error:
+        return []
+    res = []
+
+    for i in range(winreg.QueryInfoKey(handle)[0]):
+        res += [winreg.EnumKey(handle, i)]
+
+    winreg.CloseKey(handle)
+    return res
+
+
+def _read_reg_values(key, wow6432=False):
+    if isinstance(key, tuple):
+        root = key[0]
+        key = key[1]
+    else:
+        root = winreg.HKEY_CLASSES_ROOT
+
+    try:
+        if wow6432:
+            handle = winreg.OpenKey(
+                root,
+                key,
+                0,
+                winreg.KEY_READ | winreg.KEY_WOW64_32KEY
+            )
+        else:
+            handle = winreg.OpenKeyEx(root, key)
+    except winreg.error:
+        return {}
+    res = {}
+    for i in range(winreg.QueryInfoKey(handle)[1]):
+        name, value, _ = winreg.EnumValue(handle, i)
+        res[_convert_mbcs(name)] = _convert_mbcs(value)
+
+    winreg.CloseKey(handle)
+
+    return res
+
+
+def _convert_mbcs(s):
+    dec = getattr(s, "decode", None)
+    if dec is not None:
+        try:
+            s = dec("mbcs")
+        except UnicodeError:
+            pass
+    return s
+
+
+class _ModuleWrapper(object):
+
+    def __init__(self, mod):
+
+        self.__doc__ = mod.__doc__
+        self.__file__ = mod.__file__
+        self.__loader__ = mod.__loader__
+        self.__name__ = mod.__name__
+        self.__package__ = mod.__package__
+        # self.__path__ = mod.__path__
+        self.__spec__ = mod.__spec__
+        self.__original_module__ = mod
+
+        sys.modules[mod.__name__] = self
+
+    def __getattr__(self, item):
```

Comment:
`__getattribute__` is a recommended method for new-style classes. It should call `object.__getattribute__(self, item)` only when necessary so it eliminates infinite recursion.